### PR TITLE
F control extensions

### PR DESCRIPTION
**Pull request by chasglasspoole**
Adds Control Extensions Project

### CODE REVIEW

File: demoqa.testing/src/test/java/controlExtensions/MultiSelect.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+public class MultiSelect extends ControlExtensions implements IMultiSelect{
+
+	public MultiSelect(WebElement mappedElement) {
+		super(mappedElement);
+	}
+
+	@Override
+	public void selectOptions(int selections, WebDriver driver) {
+		//WebDriverWait wait = new WebDriverWait(driver, 1);
```
Comment (dentaylor) [inline demoqa.testing/src/test/java/controlExtensions/MultiSelect.java]: Remove comments

File: demoqa.testing/src/test/java/controlExtensions/MultiSelect.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		//wait.until(ExpectedConditions.elementToBeClickable(mappedElement));
+		driver.switchTo().defaultContent();
```
Comment (dentaylor) [inline demoqa.testing/src/test/java/controlExtensions/MultiSelect.java:22]: Should not be needed

File: demoqa.testing/src/test/java/controlExtensions/MultiSelect.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		for(int i = 0; i < selections; i++) {
+			Actions act = new Actions(driver);
+			act.sendKeys(Keys.ENTER).perform();
+		}
+	}
+
+	@Override
+	public String[] getSelectedOptions(WebDriver driver) {
+		driver.switchTo().defaultContent();
+		WebDriverWait wait = new WebDriverWait(driver, 1);
```
Comment (dentaylor) [inline demoqa.testing/src/test/java/controlExtensions/MultiSelect.java:33]: ImplicitWait handles presence of Elements. This should not be needed.

File: demoqa.testing/src/test/java/controlExtensions/MultiSelect.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+				wait.until(ExpectedConditions.presenceOfElementLocated(By
+				.xpath("//div[@class='css-12jo7m5']")));
+		List<WebElement> SelectedOptionsElements = driver.findElements(By.xpath("//div[@class='css-12jo7m5']"));
```
Comment (dentaylor) [inline demoqa.testing/src/test/java/controlExtensions/MultiSelect.java:36]: xpaths inside the Control Extension should be relative. Replace // with ./

File: demoqa.testing/src/test/java/controlExtensions/Slider.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+		Dimension sliderSize = mappedElement.getSize();
+		int sliderWidth = sliderSize.getWidth();
+		int startPoint = sliderWidth/2;
+		int val = Integer.parseInt(value);
+						
+		Actions moveToStart = new Actions(driver);   
+		moveToStart.moveToElement(mappedElement).moveByOffset(-startPoint, 0).click().build().perform();
+		
+		for(int i = 0; i <= val-1; i++) {
+		mappedElement.sendKeys(Keys.ARROW_RIGHT);
```
Comment (dentaylor) [inline demoqa.testing/src/test/java/controlExtensions/Slider.java]: Fix indentation

**Review by dentaylor — APPROVED** [review]

File: demoqa.testing/src/test/java/controlExtensions/MultiSelect.java (hunk 2 above)
Comment (chasglasspoole) [reply demoqa.testing/src/test/java/controlExtensions/MultiSelect.java:24]: line 24 switches to the default content so the iframe element does not intercept the click

Comment (chasglasspoole) [reply demoqa.testing/src/test/java/controlExtensions/MultiSelect.java:22]: > Should not be needed

line 24 switches to the default content so the iframe element does not intercept the click

**Pull request merged**